core/src/main/java/io/questdb/griffin/engine/functions/date/TimestampFloorFunctionFactory.java

```diff
@@ -40,28 +42,44 @@ public String getSignature() {
 
     @Override
     public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
-        final char c = args.getQuick(0).getChar(null);
+        final CharSequence str = args.getQuick(0).getStr(null);
+        char c = (str != null && str.length() == 1) ? str.charAt(0) : 0;
         switch (c) {
-            case 'd':
```

Comment:
Is there a reason to allow stride for all units except months and years ? 
If thers is a good reason then I reckon we should improve error message it'll confuse users.  
For example :
```sql
select timestamp_floor('M', now());
select timestamp_floor('y', now());
```
run fine , but the following : 
```sql
select timestamp_floor('5M', now());
select timestamp_floor('5y', now());
```
fail with  'invalid kind' message.

Reply:
The "invalid kind" error message is here on purpose.
We cannot apply this trick to months because they are not equal in time intervals.
I guess the same logic in `date_bin` constraints

Reply:
I'm not sure it's about time interval length because timestamp_floor already works for months and years already . 
I might be wrong but, assuming functions are meant for post-poch ttimestamps, it seems that  for : 
- 5y - we'd need to do year = (year / 5 ) * 5    in floorYYYY
- 5m - month = ((year * 12)/5)*5 ; year = (month/12) in floorMM 
Anyway, if you decide to keep impl as is (reject stride for month and year) then documentation needs to be clear about the limitation because error message won't say much .